### PR TITLE
ceph-rbd-mirror: use blocks for similar tasks in main.yml

### PR DESCRIPTION
**Pull request by rishabh-d-dave**

### CODE REVIEW

File: roles/ceph-rbd-mirror/tasks/main.yml [hunk 1]
```diff
@@ -12,16 +7,25 @@
...
   when: cephx
 
-- name: include start_rbd_mirror.yml
-  include_tasks: start_rbd_mirror.yml
-  when: not containerized_deployment
+- name: tasks for non-containerized deployment
+  block:
+    - name: include start_rbd_mirror.yml
+      include_tasks: start_rbd_mirror.yml
+      when: not containerized_deployment
```
Comment (guits) [inline roles/ceph-rbd-mirror/tasks/main.yml]:
```suggestion
```

File: roles/ceph-rbd-mirror/tasks/main.yml [hunk 2]
```diff
@@ -12,16 +7,25 @@
...
+
+    - name: include configure_mirroring.yml
+      include_tasks: configure_mirroring.yml
+      when:
+        - ceph_rbd_mirror_configure
+        - not containerized_deployment
```
Comment (guits) [inline roles/ceph-rbd-mirror/tasks/main.yml]:
```suggestion
```

File: roles/ceph-rbd-mirror/tasks/main.yml [hunk 3]
```diff
@@ -12,16 +7,25 @@
   include_tasks: common.yml
   when: cephx
 
-- name: include start_rbd_mirror.yml
-  include_tasks: start_rbd_mirror.yml
-  when: not containerized_deployment
+- name: tasks for non-containerized deployment
+  block:
```
Comment (guits) [inline roles/ceph-rbd-mirror/tasks/main.yml]:
```suggestion
  when: not containerized_deployment
  block:
```

File: roles/ceph-rbd-mirror/tasks/main.yml [hunk 4]
```diff
@@ -12,16 +7,25 @@
...
-  include_tasks: configure_mirroring.yml
-  when:
-    - ceph_rbd_mirror_configure
-    - not containerized_deployment
+- name: tasks for containerized deployment
+  block:
+    - name: set_fact docker_exec_cmd
+      set_fact:
+        docker_exec_cmd: "{{ container_binary }} exec ceph-rbd-mirror-{{ ansible_hostname }}"
+      when: containerized_deployment
```
Comment (guits) [inline roles/ceph-rbd-mirror/tasks/main.yml]:
```suggestion
```

File: roles/ceph-rbd-mirror/tasks/main.yml [hunk 5]
```diff
@@ -12,16 +7,25 @@
...
+        - ceph_rbd_mirror_configure
+        - not containerized_deployment
 
-- name: include configure_mirroring.yml
-  include_tasks: configure_mirroring.yml
-  when:
-    - ceph_rbd_mirror_configure
-    - not containerized_deployment
+- name: tasks for containerized deployment
+  block:
```
Comment (guits) [inline roles/ceph-rbd-mirror/tasks/main.yml]:
```suggestion
  when: containerized_deployment
  block:
```

File: roles/ceph-rbd-mirror/tasks/main.yml [hunk 6]
```diff
@@ -12,16 +7,25 @@
...
+      set_fact:
+        docker_exec_cmd: "{{ container_binary }} exec ceph-rbd-mirror-{{ ansible_hostname }}"
+      when: containerized_deployment
 
-- name: include docker/main.yml
-  include_tasks: docker/main.yml
-  when: containerized_deployment
+    - name: include docker/main.yml
+      include_tasks: docker/main.yml
+      when: containerized_deployment
```
Comment (guits) [inline roles/ceph-rbd-mirror/tasks/main.yml]:
```suggestion
```

**Comment by guits** [comment]
jenkins test pipeline

**Comment by rishabh-d-dave** [comment]
The error[[1]](https://2.jenkins.ceph.com/job/ceph-ansible-prs-dev-centos-non_container-update/121/)[[2]](https://2.jenkins.ceph.com/job/ceph-ansible-prs-dev-ubuntu-non_container-purge/78/) doesn't look relevant. So, I am trgigering those jibs again.

**Comment by rishabh-d-dave** [comment]
jenkins test dev-centos-non_container-update

**Comment by rishabh-d-dave** [comment]
jenkins test dev-ubuntu-non_container-purge

**Comment by rishabh-d-dave** [comment]
Odd; the error persisted on `dev-centos-non_container-update` but not on other jobs.

**Comment by rishabh-d-dave** [comment]
jenkins test pipeline

**Comment by rishabh-d-dave** [comment]
jenkins test dev-centos-non_container-update

**Comment by rishabh-d-dave** [comment]
second phase of testing went fine last time except for the job `dev-centos-non_container-update` -
https://2.jenkins.ceph.com/job/ceph-ansible-pipeline/3556/

**Comment by rishabh-d-dave** [comment]
jenkins test pipeline

**Comment by rishabh-d-dave** [comment]
jenkins test dev-ubuntu-container-all_daemons

**Comment by rishabh-d-dave** [comment]
jenkins test dev-ubuntu-container-all_daemons

**Pull request merged**